lib/api/core/validation/index.js

```diff
@@ -1,690 +1,680 @@
+      if (request.input.controller === 'write' && request.input.action === 'update') {
+        const updateRequest = new Request({
+          index: request.input.resource.index,
+          collection: request.input.resource.collection,
+          controller: 'read',
+          action: 'get',
+          _id: id
+        });
+        isUpdate = true;
 
-        return resolve(updateBodyClone);
-      });
-    }
-    else {
-      return resolve(request.input.body);
-    }
-  })
-  .then(documentBody => {
-    var
-      isValid = true,
-      errorMessages = verbose ? {} : [];
-
-    if (collectionSpec) {
-      if (collectionSpec.fields && collectionSpec.fields.children) {
-        try {
-          isValid = isValid && this.recurseFieldValidation(documentBody, collectionSpec.fields.children, collectionSpec.strict, errorMessages, verbose);
-        }
-        catch (error) {
-          if (error.message === 'strictness') {
-            // The strictness message can be received here only if it happens at the validation of the document's root
-            manageErrorMessage('document', errorMessages, 'The document validation is strict; it can not add unspecified sub-fields.', verbose);
-            isValid = false;
-          }
-          else {
-            throw error;
-          }
-        }
+        this.kuzzle.services.list.storageEngine.get(updateRequest).then(document => {
+          // Avoid side effects on the request during the update validation
+          updateBodyClone = _.cloneDeep(request.input.body);
+
+          _.defaultsDeep(updateBodyClone, document._source);
+
+          return resolve(updateBodyClone);
+        });
+      }
+      else {
+        return resolve(request.input.body);
       }
+    })
+      .then(documentBody => {
+        const errorMessages = verbose ? {} : [];
+        let isValid = true;
+
+        if (collectionSpec) {
+          if (collectionSpec.fields && collectionSpec.fields.children) {
+            try {
+              isValid = isValid && this.recurseFieldValidation(documentBody, collectionSpec.fields.children, collectionSpec.strict, errorMessages, verbose);
+            }
+            catch (error) {
+              if (error.message === 'strictness') {
+                // The strictness message can be received here only if it happens at the validation of the document's root
+                manageErrorMessage('document', errorMessages, 'The document validation is strict; it can not add unspecified sub-fields.', verbose);
+                isValid = false;
+              }
+              else {
+                throw error;
+              }
+            }
+          }
 
-      if (collectionSpec.validators) {
-        let filters = this.dsl.test(request.input.resource.index, request.input.resource.collection, documentBody, id);
+          if (collectionSpec.validators) {
+            const filters = this.dsl.test(request.input.resource.index, request.input.resource.collection, documentBody, id);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 125 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)